frameworks/Dart/aqueduct/server.dart

```diff
@@ -0,0 +1,228 @@
+        fortunesTemplate = new mustache.Template(template);
+      })
+    ]);
+  }
+
+  /// All routes must be configured in this method.
+  ///
+  /// This method is invoked after the constructor and before [willOpen].
+  /// All routes must be set up in this method and cannot be added after this method completes.
+  @override
+  void setupRouter(Router router) {
+    router
+        .route("/json")
+        .listen((req) async => new Response.ok({"message": "Hello, World!"})
+          ..contentType = _stripped_json
+          ..headers["date"] = new DateTime.now());
+
+    router
+        .route("/plaintext")
+        .listen((req) async => new Response.ok("Hello, World!")
+          ..contentType = _stripped_text
+          ..headers["date"] = new DateTime.now());
+
+    router.route("/db").listen((req) async {
+      World result = await getRandomWorldObject();
+      return new Response.ok(result)
+        ..contentType = _stripped_json
+        ..headers["date"] = new DateTime.now();
+    });
+
+    router.route("/queries/[:queryCount]").listen((req) async {
+      int queryCount = _min_query_count;
+      if (req.path.variables.containsKey("queryCount")) {
+        queryCount = int
+            .parse(req.path.variables["queryCount"], onError: (_) => queryCount)
+            .clamp(_min_query_count, _max_query_count);
+      }
+      List<Future> resultFutures =
+          new List.generate(queryCount, (_) => getRandomWorldObject());
+      List results = await Future.wait(resultFutures);
+      return new Response.ok(results)
+        ..contentType = _stripped_json
+        ..headers["date"] = new DateTime.now();
+    });
+
+    router.route("/updates/[:queryCount]").listen((req) async {
+      int queryCount = _min_query_count;
+      if (req.path.variables.containsKey("queryCount")) {
+        queryCount = int
+            .parse(req.path.variables["queryCount"], onError: (_) => queryCount)
+            .clamp(_min_query_count, _max_query_count);
+      }
+      List<Future> resultFutures = new List.generate(
+          queryCount,
+          (_) => getRandomWorldObject().then((World world) async {
+                Query query = new Query<World>()
+                  ..where.id = whereEqualTo(world.id)
+                  ..values.randomNumber =
+                      (_random.nextInt(_world_table_size) + 1);
+                Future<World> result = query.updateOne();
+                return await result;
+              }));
+      List results = await Future.wait(resultFutures);
+      return new Response.ok(results)
+        ..contentType = _stripped_json
+        ..headers["date"] = new DateTime.now();
+    });
+
+    router.route("/fortunes").listen((req) async {
+      Query query = new Query<Fortune>();
+      List<Fortune> results = await query.fetch();
+      results.add(new Fortune()
+        ..id = 0
+        ..message = "Additional fortune added at request time.");
+      results.sort();
```

Comment:
You can have the database do the sorting, but the additional fortune here seems to be the thing that prevents that? They want it to be an in memory sort?

Reply:
Yes, the specific requirement is to add a new model item to the results list after the query completes then sort it.